### PR TITLE
Clarified lifetime requirements of underlying resources for Music and Font

### PR DESCRIPTION
**Pull request by Bromeon**
Improves the documentation regarding the lifetime of underlying resources (file, memory buffer, input stream). This concerns the two classes `sf::Font` and `sf::Music`; the other resource classes load their contents entirely.

### CODE REVIEW

File: include/SFML/Audio/Music.hpp [hunk 1]
```diff
@@ -69,6 +69,10 @@ class SFML_AUDIO_API Music : public SoundStream
     /// See the documentation of sf::InputSoundFile for the list
     /// of supported formats.
     ///
+    /// \warning Since the music is not loaded at once but rather
+    /// streamed continuously, the file must remain accessible until
+    /// this sf::Music object loads a new music or is destroyed.
```
Comment (eXpl0it3r) [inline include/SFML/Audio/Music.hpp]: Wouldn't "...until the sf::Music object..." make more sense?

Comment (Bromeon) [inline include/SFML/Audio/Music.hpp]: Might be clearer indeed. I initially wrote "this object" (i.e. `*this`) and then added the class name for clarity.

**Comment by eXpl0it3r** [comment]
This PR has been added to my merge list, meaning it will be merged into 2.3.x soon, unless someone raises any concerns.

**Comment by eXpl0it3r** [comment]
Merged in 6de027f198e56947d89694049020de0893d374fc on branch 2.3.x

**Pull request closed without merging**